packages/nextjs/components/scaffold-eth/Contract/WriteOnlyFunctionForm.tsx

```diff
@@ -31,6 +32,9 @@ export const WriteOnlyFunctionForm = ({
 }: TWriteOnlyFunctionFormProps) => {
   const [form, setForm] = useState<Record<string, any>>(() => getInitialFormState(functionFragment));
   const [txValue, setTxValue] = useState("");
+  const { isConnected } = useAccount();
```

Comment:
Just noticed a small bug : 
If you try to disconnect from `/debug` page the modal doesn't disconnects you and also don't close 

https://user-images.githubusercontent.com/80153681/217891401-b265f494-490f-4b89-8b22-c22466eadcbd.mov

Tried debugging and found out that `useAccount` is causing this issue, removing it everything works fine. 

I think this will be fixed in https://github.com/wagmi-dev/wagmi/pull/1679
as mentioned by Carlos in https://github.com/scaffold-eth/se-2/issues/110#issuecomment-1422961150

Reply:
Are we sure this is the same issue?

I get the not closing modal part (Should be fixed when this is merged https://github.com/rainbow-me/rainbowkit/pull/1000)

But the not-disconnecting part is more concerning.

I tried some stuff and it seems to be an issue inside `RainbowKitCustomConnectButton` (If I remove it, it works). Maybe the autoconnect?

Since the error is a bit annoying and I want to merge this PR I did a (temp?) fix for it: 75ec8de2e2646e81d4ee24342c0187fa2af9a2e8